trojsten/polls/templatetags/polls_parts.py

```diff
@@ -0,0 +1,63 @@
+from django import template
+from django.conf import settings
+from django.utils import timezone
+from django.utils.translation import ngettext as _
+
+register = template.Library()
+
+
+@register.inclusion_tag("trojsten/polls/parts/progress.html", takes_context=True)
+def show_time(context, current_question):
+    start = current_question.created_date
+    end = current_question.deadline
+    full = end - start
+    remaining = end - timezone.now()
+    elapsed = full - remaining
+    if remaining.days <= settings.ROUND_PROGRESS_DANGER_DAYS:
+        progressbar_class = settings.ROUND_PROGRESS_DANGER_CLASS
+    elif remaining.days <= settings.ROUND_PROGRESS_WARNING_DAYS:
+        progressbar_class = settings.ROUND_PROGRESS_WARNING_CLASS
+    else:
+        progressbar_class = settings.ROUND_PROGRESS_DEFAULT_CLASS
+    percent = 100 * elapsed.days // full.days if full.days > 0 else 100
+    context.update(
+        {
+            "current": current_question,
+            "start": start,
+            "end": end,
+            "full": full,
+            "remaining": remaining,
+            "elapsed": elapsed,
+            "percent": percent,
+            "progressbar_class": progressbar_class,
+        }
+    )
+    return context
+
+
+@register.filter
+def progress_time(delta):
+    if delta.days >= 1:
+        count = delta.days
+        return _("%(count)d day", "%(count)d days", count) % {"count": count}
+    elif delta.seconds // 3600 >= 1:
+        count = delta.seconds // 3600
+        return _("%(count)d hour", "%(count)d hours", count) % {"count": count}
+    elif delta.seconds // 60 >= 1:
+        count = delta.seconds // 60
+        return _("%(count)d minute", "%(count)d minutes", count) % {"count": count}
+    else:
+        count = delta.seconds
+        return _("%(count)d second", "%(count)d seconds", count) % {"count": count}
+
+
+@register.filter
+def progress_time_precision(delta):
+    if delta.days >= 1:
+        return "DAY"
+    elif delta.seconds // 3600 >= 1:
+        return "HOUR"
+    elif delta.seconds // 60 >= 1:
+        return "MINUTE"
+    else:
+        return "SECOND"

```

Comment:
Tieto veci su uplne rovnake ako v statements. Ai by bolo lepsie to presunut niekam asi do utils.
Este lepsie by bolo presunut cely progressbar niekam do utils aj s templatami a vsetkym a pouzit to iste aj tu aj v statements.

Reply:
Mam to robit v tomto pull requeste? Nehodilo by sa to zvlast?

Reply:
Asi je lepsie zvlast, ale ako chces. Som okej aj stym ze to bude v tomto PR.

Ked to spravis, malo by to posunut test coverage tohto PR do zelenych cisel.

Reply:
Neviem vsak, ako to spravit, kedze v statements k tomu progressbaru prislucha aj ten javascript, ktory progressbar updatuje. Ten skript je tusim roundprogressbar.js v static/js/, ale pristupuje k nejakej samostatnej url, ktora je zavisla od ulohy, z ktorej cita cas konca. Bolo by nejak do toho skriptu treba dostat udaj, ze ide o ulohu + id ulohy.

Reply:
Ak chces, tak nateraz mozes pridat nejaky boolean flag, ktory prida ten js iba pre statements. Plus vyrob issue na to aby sa to spravilo poriadne.

Ale podla mna spravit to poriadne neni az taky problem. Podobne ako sa teraz dava id ulohy do `data-id`, tak sa da pridat refresh url do `data-refreshurl` alebo nieco take. Potom sa da pridat podpora pre refresh aj pre anketovy progressbar. Ale pokojne to mozme spravit v nasledujucom PR.

Reply:
Premiestnil som tieto dve funkcie do trojsten/utils/utils.py.
Teraz mi to ale hadze TemplateSyntaxError, invalid filter progress_time_precision.
Teda tieto funkcie, ktore maju byt filtrami, nepozna.
V utils to vyzera takto:

```
register = template.Library()
@register.filter
def progress_time(delta):
    # telo funkcie
@register.filter
def `progress_time_precision(delta):`
    # telo funkcie
```
Velmi neviem, ako to mam spravit. Tie funkcie asi nejak treba dekorovat tym register.filter zvlast v statements_parts a zvlast v polls_parts, no neviem dekorovat hotovu funkciu a neviem ani, ci to pomoze.

Reply:
Tu tretiu funkciu (show_time/show_progress) sa tam ani nejdem pokusat premiestnit, pretoze je dost odlisna v statements a v polls. Odlisny je sposob ziskavania zaciatku a konca, v statements je navyse parameter results, ktory sa do posiela do context-u. Je to sice podobny kod, ale nie uplny duplikat.

Reply:
Obisiel som ten problem tak, ze v *_parts.py som nechal hlavicku funkcie, v tele volam funkciu z utils.